conclave-enclave/src/main/kotlin/com/r3/conclave/enclave/internal/NativeEnclaveHostInterface.kt

```diff
@@ -0,0 +1,121 @@
+package com.r3.conclave.enclave.internal
+
+import com.r3.conclave.common.EnclaveStartException
+import com.r3.conclave.common.internal.*
+import com.r3.conclave.mail.MailDecryptionException
+import com.r3.conclave.utilities.internal.getAllBytes
+import java.nio.ByteBuffer
+import java.util.*
+
+typealias StackFrame = CallInterfaceStackFrame<HostCallType>
+
+/**
+ * This class is the implementation of the [EnclaveHostInterface] for native enclaves.
+ * It has three jobs:
+ *  - Serve as the endpoint for calls to make to the host, see [com.r3.conclave.common.internal.CallInterface]
+ *  - Route calls from the host to the appropriate enclave side call handler, see [com.r3.conclave.common.internal.CallInterface]
+ *  - Handle the low-level details of the messaging protocol (ecalls and ocalls).
+ */
+class NativeEnclaveHostInterface : EnclaveHostInterface() {
+    /** In release mode we want to sanitise exceptions to prevent leakage of information from the enclave */
+    var sanitiseExceptions: Boolean = false
+
+    /**
+     * Each thread has a lazily created stack which contains a frame for the currently active host call.
+     * When a message arrives from the host, this stack is used to associate the return value with the corresponding call.
+     */
+    private val threadStacks = ThreadLocal<Stack<StackFrame>>()
+    private val stack: Stack<StackFrame>
+        get() {
+            if (threadStacks.get() == null) {
+                threadStacks.set(Stack<StackFrame>())
+            }
+            return threadStacks.get()
+        }
```

Comment:
This can be simplified if you can create your thread local using `withInitial` factory method.